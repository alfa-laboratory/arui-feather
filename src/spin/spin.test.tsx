/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { mount } from 'enzyme';

import Spin from './spin';
import ThemedSpin from './index';

describe('spin', () => {
    it('should render without problems', () => {
        const spin = mount(<Spin />);

        expect(spin).toMatchSnapshot();
    });

    it('should set class `spin_visible` when visible=true', () => {
        const spin = mount(<Spin visible={ true } />);

        expect(spin.getDOMNode().className).toContain('spin_visible');
    });

    it('should render themed component without problems', () => {
        const spin = mount(<ThemedSpin>Themed label test</ThemedSpin>);

        expect(spin).toMatchSnapshot();
    });

    it('should contains theme class', () => {
        const spin = mount(<ThemedSpin />);

        expect(spin.getDOMNode().className).toContain('spin_theme_alfa-on-white');
    });
});
