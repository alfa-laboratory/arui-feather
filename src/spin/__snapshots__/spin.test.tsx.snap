// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spin should render themed component without problems 1`] = `
<ForwardRef(ThemedComponent(Spin))>
  <Spin
    size="m"
    theme="alfa-on-white"
    visible={false}
  >
    <span
      className="spin spin_size_m spin_theme_alfa-on-white"
    />
  </Spin>
</ForwardRef(ThemedComponent(Spin))>
`;

exports[`spin should render without problems 1`] = `
<Spin
  size="m"
  visible={false}
>
  <span
    className="spin spin_size_m"
  />
</Spin>
`;
